pkg/api/types.go

```diff
@@ -1829,3 +1811,7 @@ func FormatProdFQDNByLocation(fqdnPrefix string, location string, cloudName stri
 	FQDNFormat := AzureCloudSpecEnvMap[targetEnv].EndpointConfig.ResourceManagerVMDNSSuffix
 	return fmt.Sprintf("%s.%s."+FQDNFormat, fqdnPrefix, location)
 }
+
+func (p *Properties) isKubernetesAndNotAzureStack() bool {
+	return p.OrchestratorProfile != nil && p.OrchestratorProfile.KubernetesConfig != nil && !p.IsAzureStackCloud()
```

Comment:
I assume this is already unit tested through the other functions' unit tests?

Reply:
We should see failed tests if this function is not doing what is supposed to do

Reply:
Is there a reason not to just make a method with a simpler name like "`isAzureStack()`" ? Then we can employ `!isAzureStack()` if we want to validate the NOT condition.

(Also as AKS-Engine + Azure Stack is very unlikely to ever be anything other than Kubernetes, I think we can hide the "is kubernetes" logic behind the method, but we don't need to advertise that in the method name)

Reply:
I had suggested this explicit function name, but `isAzureStack` probably covers it.

Reply:
We want to check for non-nil orcherstrator profile values regardless of Azure Stack so I don't think we should embed that in the IsAzureStack func. If later we want to re-enable some or all of these addons for Azure Stack, we will still want to keep the `isKubernetes` validation.

Reply:
Renamed function and kept nil checks.